internal/regexp/regexp.go

```diff
@@ -0,0 +1,76 @@
+package regexp
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+// Compile takes a regular expression and a custom lookup struct that should match the named groups
+// in the regexp.
+//
+// `lookup` should always be a pointer to a struct with only public `int` properties.
+//
+// Example:
```

Comment:
Would be nice with an `ExampleCompile()` test function instead as this comment is not typed checked :)

Reply:
Nice suggestion 💪